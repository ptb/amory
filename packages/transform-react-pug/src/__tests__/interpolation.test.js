import testHelper from './test-helper';

testHelper(__dirname + '/interpolation.input.js');
