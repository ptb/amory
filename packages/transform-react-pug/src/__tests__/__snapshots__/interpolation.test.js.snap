// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`JavaScript output: transformed source code 1`] = `
"function Person({ name }) {
  return <div className=\\"person person__profile\\"><h2>{['My name is ', name].join('')}</h2></div>;
}

module.exports = new function () {
  this.message = 'Hello everyone';
  this.names = ['jack', 'john'];
  const showFirstPerson = true;

  return <div className=\\"people people__container\\"><h1>{this.message}</h1><p>Here is a list of people that you might want to follow,{\\"\\\\n\\"}{['one being ', this.names[1], '!'].join('')}</p>{showFirstPerson ? <Person name={this.names[0]} key=\\"pug:0:0\\" /> : undefined}{this.names.map((name, key) => <p key={key}>{['Hi ', name, ', bye ', name].join('')}</p>)}</div>;
}();"
`;

exports[`html output: generated html 1`] = `
<div
  className="people people__container"
>
  <h1>
    Hello everyone
  </h1>
  <p>
    Here is a list of people that you might want to follow,
    

    one being john!
  </p>
  <div
    className="person person__profile"
  >
    <h2>
      My name is jack
    </h2>
  </div>
  <p>
    Hi jack, bye jack
  </p>
  <p>
    Hi john, bye john
  </p>
</div>
`;

exports[`static html output: static html 1`] = `
"<div class=\\"people people__container\\"><h1>Hello everyone</h1><p>Here is a list of people that you might want to follow,
one being john!</p><div class=\\"person person__profile\\"><h2>My name is jack</h2></div><p>Hi jack, bye jack</p><p>Hi john, bye john</p></div>"
`;
